packages/langium/src/grammar/validation/validator.ts

```diff
@@ -411,6 +413,45 @@ export class LangiumGrammarValidator {
         }
     }
 
+    checkEmptyParserRule(parserRule: ast.ParserRule, accept: ValidationAcceptor): void {
+        // Rule body needs to be set.
+        // Entry rules and fragments may consume no input.
+        if (!parserRule.definition || parserRule.entry || parserRule.fragment) {
+            return;
+        }
+
+        // Process definition; report problem for the rule itself.
+        const mayConsumeEmpty = (element: ast.AbstractElement): boolean => {
+            // First, check cardinality of the element.
+            if (element.cardinality === '?' || element.cardinality === '*') {
+                return true;
+            }
+            // Actions themselves count as non-optional.
+            if (ast.isAction(element)) {
+                return false;
+            }
+            // Assignments with ?= consume nothing, groups without + too.
+            if (ast.isAssignment(element)) {
+                return element.operator === '?=';
```

Comment:
Suggestion: Every assignment always consumes something (as long as its cardinality isn't `?` or `*`):

```
A: value?='test';
```

Consumes the `'test'` keyword input. You can just remove the case completely, since it should always return `false` for assignments.

Reply:
Case removed.